presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -377,6 +377,7 @@ public TaskStats getTaskStats()
                 completedDrivers,
                 cumulativeMemory.get(),
                 succinctBytes(memoryReservation.get()),
+                succinctBytes(queryContext.getPeakReservedMemory()),
```

Comment:
Don't understand this at all - why is the task constructing it's stats using the queryContext?